components/script/dom/event.rs

```diff
@@ -690,12 +690,15 @@ fn inner_invoke(
             object.remove_listener_if_once(&event.type_(), &event_listener);
         }
 
-        // Step 2.6-2.8
-        // FIXME(#25478): we need to get the global that the event
-        // listener is going to be called on, then if it's a Window
-        // set its .event to the event, remembering the previous
-        // value of its .event. This allows events to just use
-        // the word "event" instead of taking the event as an argument.
+        // Step 2.6
+        let global = listener.associated_global();
+
+        // Step 2.7-2.8
+        let current_event = if let Some(window) = global.downcast::<Window>() {
```

Comment:
i would perhaps call this `last_event` or something?

Reply:
I'm going with the name from the spec.